nsq/message.py

```diff
@@ -51,13 +51,22 @@ class Message(event.EventedMixin):
     def __init__(self, id, body, timestamp, attempts):
         self._async_enabled = False
         self._has_responded = False
+
         self.id = id
         self.body = body
         self.timestamp = timestamp
         self.attempts = attempts
+        self.timed_out = False
 
         super(Message, self).__init__()
 
+    def is_alive(self):
+        """
+        Indicates whether or not this :class:`nsq.message.Message` instance is timed out based on the
+            msg_timeout parameter.
+        """
+        return not self.timed_out
```

Comment:
We may want to use `not self.timed_out and not self._has_responded` here for a more comprehensive check